x-pack/plugins/infra/public/alerting/inventory/components/expression_chart.tsx

```diff
@@ -0,0 +1,237 @@
+  sourceId: string;
+}
+
+export const ExpressionChart: React.FC<Props> = ({
+  expression,
+  filterQuery,
+  nodeType,
+  sourceId,
+}) => {
+  const timerange = useMemo(
+    () => ({
+      interval: `${expression.timeSize || 1}${expression.timeUnit}`,
+      from: moment()
+        .subtract((expression.timeSize || 1) * 20, expression.timeUnit)
+        .valueOf(),
+      to: moment().valueOf(),
+      forceInterval: true,
+      ignoreLookback: true,
+    }),
+    [expression.timeSize, expression.timeUnit]
+  );
+
+  const buildCustomMetric = (metric: any) => ({
+    ...metric,
+    type: 'custom' as SnapshotMetricType,
+  });
+
+  const { loading, nodes } = useSnapshot(
+    filterQuery,
+    expression.metric === 'custom'
+      ? [buildCustomMetric(expression.customMetric)]
+      : [{ type: expression.metric }],
+    [],
+    nodeType,
+    sourceId,
+    0,
+    '',
+    '',
+    true,
+    timerange
+  );
+
+  const { uiSettings } = useKibanaContextForPlugin().services;
+
+  const metric = {
+    field: expression.metric,
+    aggregation: 'avg' as MetricsExplorerAggregation,
+    color: Color.color0,
+  };
+  const isDarkMode = uiSettings?.get('theme:darkMode') || false;
+  const dateFormatter = useMemo(() => {
+    const firstSeries = nodes[0]?.metrics[0]?.timeseries;
+    const firstTimestamp = first(firstSeries?.rows)?.timestamp;
+    const lastTimestamp = last(firstSeries?.rows)?.timestamp;
+
+    if (firstTimestamp == null || lastTimestamp == null) {
+      return (value: number) => `${value}`;
+    }
+
+    return niceTimeFormatter([firstTimestamp, lastTimestamp]);
+  }, [nodes]);
+
+  /* eslint-disable-next-line react-hooks/exhaustive-deps */
+  const yAxisFormater = useCallback(
+    createInventoryMetricFormatter(
+      expression.metric === 'custom'
+        ? buildCustomMetric(expression.customMetric)
+        : { type: expression.metric }
+    ),
+    [expression.metric]
+  );
+
+  if (loading || !nodes) {
+    return <LoadingState />;
+  }
+
+  const criticalThresholds = expression.threshold.slice().sort();
+  const warningThresholds = expression.warningThreshold?.slice().sort() ?? [];
+  const thresholds = [...criticalThresholds, ...warningThresholds].sort();
+
+  // Creating a custom series where the ID is changed to 0
+  // so that we can get a proper domian
+  const firstSeries = nodes[0]?.metrics[0]?.timeseries;
+  if (!firstSeries || !firstSeries.rows || firstSeries.rows.length === 0) {
+    return <NoDataState />;
+  }
+
+  const thresholdFormatter = (value: number | undefined) => {
+    const metricFormatter = METRIC_FORMATTERS[expression.metric];
+
+    if (!value || !metricFormatter) {
+      return value;
+    }
+
+    switch (metricFormatter.formatter) {
+      case InfraFormatterType.percent:
+        return value / 100;
+      case InfraFormatterType.bits:
+        return value / 1000;
+      default:
+        return value;
+    }
+  };
+
+  const series = {
+    ...firstSeries,
```

Comment:
We should set the `id` field to use `nodes[0]?.name` so when the chart is rendered it will put the host name in the label instead of `cpu`. 

![image](https://user-images.githubusercontent.com/41702/109198427-71359b80-775b-11eb-85b8-d13a1c615eb1.png)

Reply:
Done!